util/feeding.py

```diff
@@ -1,195 +1,434 @@
 import pandas
 import tensorflow as tf
+import time
+import Queue
 
-from threading import Thread
+from threading import Thread, Lock
 from math import ceil
 from six.moves import range
 from util.audio import audiofile_to_input_vector
-from util.gpu import get_available_gpus
 from util.text import ctc_label_dense_to_sparse, text_to_char_array
+from util.log import Logger
+
+log = Logger('feeding', 'Feeding')
+
+class DataSet(object):
+    '''
+    Represents a collection of audio samples and their respective transcriptions.
+    Takes a set of CSV files produced by importers in /bin.
```

Comment:
nit: probably `bin/` instead of `/bin` ;)

Reply:
:-)